JS and R (#123)
r-hub/blog: content/post/2020-07-27-js-r/index.Rmd

--- 
slug: js-r 
title: "JavaScript for the R package developer" 
authors: 
- Maëlle Salmon
- Garrick Aden-Buie
date: "2020-08-20" 
tags: 
- package development 
- JS
output: hugodown::hugo_document
---

```{r setup, include=FALSE}

# Options to have images saved in the post folder
# And to disable symbols before output
knitr::opts_chunk$set(fig.path = "", comment = "")
# knitr hook to make images output use Hugo options
knitr::knit_hooks$set(
  plot = function(x, options) {
    hugoopts <- options$hugoopts
    paste0(
      "{{<figure src=",
      '"', x, '" ',
      if (!is.null(hugoopts)) {
        glue::glue_collapse(
          glue::glue('{names(hugoopts)}="{hugoopts}"'),
          sep = " "
        )
      },
      ">}}\n"
    )
  }
)

```

JS and R, what a clickbait!
Come for JS, stay for our posts about [Solaris](/2020/05/14/checking-your-r-package-on-solaris/) and [WinBuilder](/2020/04/01/win-builder/). :wink:
No matter how strongly you believe in JavaScript being the language of the future (see below), you might still gain from using it in your R practice, be it back-end or front-end.

`r hugodown::embed_tweet("1272595824112029696")`

> In this blog post, we shall share a roundup of resources around JavaScript for R package developers.

Reviewer: ```suggestion
In this blog post, [Garrick Aden-Buie](https://www.garrickadenbuie.com) and I share a roundup of resources around JavaScript for R package developers.
```

> > All the smart JavaScript content of this post can be attributed to [Garrick Aden-Buie](https://www.garrickadenbuie.com/). :pray:

Reviewer: ```suggestion
```

I appreciate the call-out, but this was very much a collaborative effort and I'd be more comfortable with just a link to my page. :relaxed:

> `---

Reviewer: Is this a stray <code>`</code>?

In this post we went over some resources useful to R package developers looking to use JavaScript code in the backend or docs of their packages, or to help others use JavaScript dependencies. Do not hesitate to share more links or experience in the comments below!

[^size]: For instance, remember that [for CRAN](https://cran.r-project.org/web/packages/policies.html), "*neither data nor documentation should exceed 5MB*".

> `

Reviewer: Also this one?
